tools/common_lib/src/gemm.h

```diff
@@ -944,15 +950,40 @@ class GemmUmdD3d12Dispatcher : public GemmBaseDispatcher
     {
         using namespace dnnl_utils;
 
-        input_a_memory_desc_ = to_dnnl_mem_desc(params_.shape_a, params_.layout, params_.dt);
-        const auto input_b_memory_desc = to_dnnl_mem_desc(params_.shape_b, params_.b_managed ? DataLayout::eWeightsLayoutStart : params_.layout, params_.dt);
+        //input_a_memory_desc_ = to_dnnl_mem_desc(params_.a_transposed ? TensorShape{ params_.shape_a.n, params_.shape_a.c, params_.shape_a.w, params_.shape_a.h } : params_.shape_a, params_.layout, params_.dt);
+        if (params_.a_transposed)
+        {
+            input_a_memory_desc_ = to_dnnl_mem_desc(params_.shape_a, params_.layout, params_.dt);
```

Comment:
line 956 can be moved above the if condition, so "else" condition can be removed.
The same for below tensor b